### ISSUE TITLE
Is the tensorflow-probability release no longer compatible with JAX

### ISSUE THREAD

**Issue by NeilGirdhar** [tensorflow/probability#1289]
```
$ pip install tensorflow_probability
Collecting tensorflow_probability
  Using cached tensorflow_probability-0.12.1-py2.py3-none-any.whl (4.8 MB)
Requirement already satisfied: gast>=0.3.2 in /home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.9/lib/python3.9/site-packages (from tensorflow_probability) (0.4.0)
Requirement already satisfied: dm-tree in /home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.9/lib/python3.9/site-packages (from tensorflow_probability) (0.1.5)
Requirement already satisfied: decorator in /home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.9/lib/python3.9/site-packages (from tensorflow_probability) (5.0.3)
Requirement already satisfied: numpy>=1.13.3 in /home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.9/lib/python3.9/site-packages (from tensorflow_probability) (1.20.2)
Requirement already satisfied: cloudpickle>=1.3 in /home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.9/lib/python3.9/site-packages (from tensorflow_probability) (1.6.0)
Requirement already satisfied: six>=1.10.0 in /home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.9/lib/python3.9/site-packages (from tensorflow_probability) (1.15.0)
Installing collected packages: tensorflow-probability
Successfully installed tensorflow-probability-0.12.1
```
```
pytest tests                                                                                                                          ✔  efax-ZssyUsLU-py3.9  
ImportError while loading conftest '/home/neil/src/efax/tests/conftest.py'.
tests/conftest.py:9: in <module>
    from .create_info import create_infos
tests/create_info.py:8: in <module>
    from efax import (BernoulliEP, BernoulliNP, BetaEP, BetaNP, ChiSquareEP, ChiSquareNP,
efax/__init__.py:1: in <module>
    from .bernoulli import *
efax/bernoulli.py:11: in <module>
    from .beta import BetaNP
efax/beta.py:9: in <module>
    from .dirichlet_common import DirichletCommonEP, DirichletCommonNP
efax/dirichlet_common.py:11: in <module>
    from .exp_to_nat import ExpToNat
efax/exp_to_nat.py:12: in <module>
    from .expectation_parametrization import ExpectationParametrization
efax/expectation_parametrization.py:7: in <module>
    from .natural_parametrization import NaturalParametrization
efax/natural_parametrization.py:13: in <module>
    from .parametrization import Parametrization
efax/parametrization.py:12: in <module>
    from .tools import tree_dot_final
efax/tools.py:38: in <module>
    ive = tfp.math.bessel_ive
../../.cache/pypoetry/virtualenvs/efax-ZssyUsLU-py3.9/lib/python3.9/site-packages/tensorflow_probability/python/internal/lazy_loader.py:57: in __getattr__
    module = self._load()
../../.cache/pypoetry/virtualenvs/efax-ZssyUsLU-py3.9/lib/python3.9/site-packages/tensorflow_probability/python/internal/lazy_loader.py:44: in _load
    module = importlib.import_module(self.__name__)
../../.cache/pypoetry/virtualenvs/efax-ZssyUsLU-py3.9/lib/python3.9/site-packages/tensorflow_probability/substrates/jax/__init__.py:44: in <module>
    from tensorflow_probability.substrates.jax import bijectors
../../.cache/pypoetry/virtualenvs/efax-ZssyUsLU-py3.9/lib/python3.9/site-packages/tensorflow_probability/substrates/jax/bijectors/__init__.py:23: in <module>
    from tensorflow_probability.substrates.jax.bijectors.absolute_value import AbsoluteValue
../../.cache/pypoetry/virtualenvs/efax-ZssyUsLU-py3.9/lib/python3.9/site-packages/tensorflow_probability/substrates/jax/bijectors/absolute_value.py:23: in <module>
    from tensorflow_probability.substrates.jax.bijectors import bijector
../../.cache/pypoetry/virtualenvs/efax-ZssyUsLU-py3.9/lib/python3.9/site-packages/tensorflow_probability/substrates/jax/bijectors/bijector.py:35: in <module>
    from tensorflow_probability.substrates.jax.math import gradient
../../.cache/pypoetry/virtualenvs/efax-ZssyUsLU-py3.9/lib/python3.9/site-packages/tensorflow_probability/substrates/jax/math/__init__.py:23: in <module>
    from tensorflow_probability.substrates.jax.math import psd_kernels
../../.cache/pypoetry/virtualenvs/efax-ZssyUsLU-py3.9/lib/python3.9/site-packages/tensorflow_probability/substrates/jax/math/psd_kernels/__init__.py:22: in <module>
    from tensorflow_probability.substrates.jax.math.psd_kernels.exp_sin_squared import ExpSinSquared
../../.cache/pypoetry/virtualenvs/efax-ZssyUsLU-py3.9/lib/python3.9/site-packages/tensorflow_probability/substrates/jax/math/psd_kernels/exp_sin_squared.py:26: in <module>
    from tensorflow_probability.substrates.jax.math.psd_kernels.internal import util
../../.cache/pypoetry/virtualenvs/efax-ZssyUsLU-py3.9/lib/python3.9/site-packages/tensorflow_probability/substrates/jax/math/psd_kernels/internal/util.py:110: in <module>
    def sqrt_with_finite_grads(x, name=None):
../../.cache/pypoetry/virtualenvs/efax-ZssyUsLU-py3.9/lib/python3.9/site-packages/tensorflow_probability/python/internal/backend/jax/_utils.py:67: in wrap
    return new_fn(*args, **kwargs)
../../.cache/pypoetry/virtualenvs/efax-ZssyUsLU-py3.9/lib/python3.9/site-packages/tensorflow_probability/python/internal/backend/jax/ops.py:422: in _custom_gradient
    ???
E   AttributeError: module 'jax' has no attribute 'custom_transforms'
```
This file does not appear in your repository, which suggests that it is automatically generated.
```
$ pip install tfp_nightly
Collecting tfp_nightly
  Downloading tfp_nightly-0.13.0.dev20210331-py2.py3-none-any.whl (5.2 MB)
     |████████████████████████████████| 5.2 MB 4.9 MB/s 
Requirement already satisfied: dm-tree in /home/neil/.cache/pypoetry/virtualenvs/efax-ZssyUsLU-py3.9/lib/python3.9/site-packages (from tfp_nightly) (0.1.5)
Requirement already satisfied: decorator in /home/neil/.cache/pypoetry/virtualenvs/efax-ZssyUsLU-py3.9/lib/python3.9/site-packages (from tfp_nightly) (5.0.3)
Requirement already satisfied: cloudpickle>=1.3 in /home/neil/.cache/pypoetry/virtualenvs/efax-ZssyUsLU-py3.9/lib/python3.9/site-packages (from tfp_nightly) (1.6.0)
Requirement already satisfied: gast>=0.3.2 in /home/neil/.cache/pypoetry/virtualenvs/efax-ZssyUsLU-py3.9/lib/python3.9/site-packages (from tfp_nightly) (0.4.0)
Requirement already satisfied: six>=1.10.0 in /home/neil/.cache/pypoetry/virtualenvs/efax-ZssyUsLU-py3.9/lib/python3.9/site-packages (from tfp_nightly) (1.15.0)
Requirement already satisfied: numpy>=1.13.3 in /home/neil/.cache/pypoetry/virtualenvs/efax-ZssyUsLU-py3.9/lib/python3.9/site-packages (from tfp_nightly) (1.20.2)
Installing collected packages: tfp-nightly
Successfully installed tfp-nightly-0.13.0.dev20210331
```
After which the tests run normally.

What is going on?  Do we need to wait for the next version of tensorflow-probability for JAX compatibility?

**Comment by axch** [comment]
IIUC, the latest JAX broke our stable release by removing
`custom_transforms`.  The nightly TFP release should work; as should
pinning jax==0.2.11

**Comment by NeilGirdhar** [comment]
Okay, no problem, I was just wondering why it was broken.

**Comment by brianwa84** [comment]
There was a jax release that broke stable tfp. You can either use
tfp-nightly or jax==0.2.11

**Comment by NeilGirdhar** [comment]
Yup, thanks!  I'll just use nightly until your next release :smile:

**Comment by brianwa84** [comment]
I released 0.12.2 just now, so `pip install tensorflow-probability[jax]` should work again.

**Comment by tigerneil** [comment]
> I released 0.12.2 just now, so `pip install tensorflow-probability[jax]` should work again.

I find it works for me.
